## Lectures

### Machine Learning in Bioinformatics - FU Berlin

Machine Learning Models:
- From Linear Regression to Kernel Regression [[Slides]](data/mlb-slides/08_-_kernel_regression.pdf)
- Support Vector Machines [[Slides]](data/mlb-slides/09_-_svm.pdf)
- From Linear and Logistic Regression to Artificial Neural Networks [[Slides]](data/mlb-slides/10_-_ann.pdf)
- Neural Network Architectures [[Slides]](data/mlb-slides/11_-_ann_architectures.pdf)
- Graph Neural Networks [[Slides]](data/mlb-slides/14_-_gnn.pdf)
- Invertible Neural Networks [[Slides]](data/mlb-slides/15_-_inn.pdf)

Feature and Model Selection, Bias-Variance Tradeoff, Regularization, Model Complexity, Double Descent:
- Model Selection [[Slides]](data/mlb-slides/12_-_model_selection.pdf)
- Feature Selection [[Slides]](data/mlb-slides/13_-_feature_selection.pdf)

Model Evaluation and Explainability:
- Benchmarking [[Slides]](data/mlb-slides/02_-_benchmarking.pdf)
- Explainable AI [[Slides]](data/mlb-slides/16_-_xai.pdf)

Statistics background:
- Decision Making under Uncertainty [[Slides]](data/mlb-slides/17_-_decision_making.pdf)

## Software projects

### Numerics, Statistics, ML

- Leapfrog regularization for pytorch
  [[Github]](https://github.com/pbenner/pyleapfrog)
- Numerical optimization, statistical estimation, and automatic differentiation library
  [[Github]](https://github.com/pbenner/autodiff)
- Variational inference on factor graphs
  [[Github]](https://github.com/pbenner/variational-fg)
- GNU-R Gaussian process regression library
  [[Github]](https://github.com/pbenner/gp.regression)
- High-dimensional logistic regression with k-mers
  [[Github]](https://github.com/pbenner/kmerLr)
- Double descent phenomena
  [[Github]](https://github.com/pbenner/double-descent)
- Compute performance measures of classification results
  [[Github]](https://github.com/pbenner/classifierPerformance)
- Active learning library
  [[Github]](https://github.com/pbenner/adaptive-sampling)

### Bioinformatics

- Bioinformatics library
  [[Github]](https://github.com/pbenner/gonetics)
- Supra-Bayesian genome segmentation method
  [[Github]](https://github.com/pbenner/modhmm)
- Statistical analysis of whole genome sequencing data
  [[Github]](https://github.com/pbenner/ngstat)
- Java library for reading bigWig files (port from gonetics)
  [[Github]](https://github.com/pbenner/jigwig)
- Identification of transcription factor binding motifs from ChIP-seq experiments. Estimation and computation of geometric means and medians of phylogenetic trees
  [[Github]](https://github.com/pbenner/tfbayes)
- Prolog implementation of all steps involved in translation and transcription of the genetic code. The program allows to explore alternative genetic codes by manipulating individual components of the transcription and translation mechanisms
  [[Github]](https://github.com/pbenner/xenocode)

### Other

- Implementation of basic cryptographic methods
  [[Github]](https://github.com/pbenner/gocrypt)
- Go threadpool library that allows nested queuing
  [[Github]](https://github.com/pbenner/threadpool)

## Publications

- A. Naik, C. Ertural, N. Dhamrait, P, Benner, J. George. *A Quantum-Chemical Bonding Database for Solid-State Materials.* arXiv preprint arXiv:2304.02726 (2023).
- B. R. Pauw, S. Laskina, A. Naik, G. J. Smales, J. George, I. Breßler, P. Benner. *``Ultima Ratio'': Simulating wide-range X-ray scattering and diffraction.* arXiv preprint arXiv:2303.13435 (2023)
- Y. Nowatzky, P. Benner, K. Reinert, T. Muth. *Mistle: bringing spectral library predictions to metaproteomics with an efficient search index.* bioRxiv (2022)
- P. Benner, and M. Vingron. *Quantifying the Tissue-Specific Regulatory Information within Enhancer and Promoter DNA Sequences.* NAR Genomics and Bioinformatics 3.4 (2021)
- P. Benner. *Computing leapfrog regularization paths with applications to large-scale k-mer logistic regression.* Journal of Computational Biology 28.6 (2021): 560-569.
- P. Benner, and M. Vingron. *ModHMM: A modular supra-Bayesian genome segmentation method.* Journal of Computational Biology 27.4 (2020): 442-457.
- T. Zehnder, P. Benner, and M. Vingron. *Predicting enhancers in mammalian genomes using supervised hidden Markov models.* BMC Bioinformatics 20(1):157, 2019
- A. Ramisch, V. Heinrich, ..., P. Benner, ... *CRUP: A comprehensive framework to predict condition-specific regulatory units.* Genome Biol 20, 227 (2019) doi:10.1186/s13059-019-1860-7
- S. Schöne, M. Bothe, E. Einfeldt, M. Borschiwer and P. Benner, M. Vingron, Martin, M. Thomas-Chollier, and S. Meijsing. *Synthetic STARR-seq reveals how DNA shape and sequence modulate transcriptional output and noise*. PLoS Genetics, 14(11):e1007793, 2018
- P. Benner, M. Bacák, and P.-Y. Bourguignon. *Point estimates in phylogenetic reconstructions*. Bioinformatics, 30(17):i534-i540, 2014.
- P. Benner and S. Poppe. *Stochastische Prozesse und Bayessches Schätzen*.
In Nicole J. Saam and Norman Braun, editors, Handbuch Modellbildung und Simulation in den Sozialwissenschaften, Springer, 2014.
- S. Poppe, P. Benner, and T. Elze. *A predictive approach to nonparametric inference for adaptive sequential sampling of psychophysical experiments*. Journal of Mathematical Psychology, 56(3):179–195, 2012.
- R. Stoop, S. Martignoli, P. Benner, R. Stoop, and Y. Uwate. *Shrimps: Occurrence, scaling and relevance*. International Journal of Bifurcation and Chaos, 22(10), 2012.
- R. Stoop, P. Benner, and Y. Uwate. *Real-world existence and origins of global shrimp organization on spirals*. Physical Review Letters, 105(7):074102, 2010.

## Statistics Resources

- Research Articles by E. T. Jaynes [[Link]](https://bayes.wustl.edu/etj/node1.html)
- Introduction to Statistical Learning with Applications in GNU-R/Python [[Link]](https://www.statlearning.com/)
